Init

MichaelMcKinsey1 opened the pull request:
# Summary (Write a short headline summary of PR)

- This PR is a (refactoring, bugfix, feature, something else)
- It does the following (modify list as needed):
  - Modifies/refactors (class or method) (how?)
  - Fixes (issue number(s))
  - Adds (specific feature) at the request of (project or person)

MichaelMcKinsey1 commented:
Ignore the tuning commits, they will go away once #333 is merged

MichaelMcKinsey1 commented on src/common/KernelBase.hpp:
```diff
@@ -308,6 +341,7 @@ class KernelBase
     MPI_Barrier(MPI_COMM_WORLD);
 #endif
     timer.start();
+    CALI_START;
```
I put `CALI_START` after and `CALI_STOP` before the existing timers. Is this acceptable? I assume if someone turns caliper on they would be looking at those timings so they should be more accurate and in the opposite case caliper would be off so it shouldn't impact the original timers that much. Not sure how much impact this has.

rhornung67 replied:
I think that choice makes most sense.

rhornung67 replied:
If Caliper is off, do the macros become no-ops with a simple run time conditional evaluation?

MichaelMcKinsey1 replied:
Yes. If RAJAPerf is not built with caliper, `RAJA_PERFSUITE_USE_CALIPER=OFF`, the macros will be defined as [here](https://github.com/LLNL/RAJAPerf/pull/331/files#diff-f1b7ede2b59bf264efbe43bc92349deb28c8f49dc87efa11b4882ab6805beb24R61), so the only operation will be the `#if defined(RAJA_PERFSUITE_USE_CALIPER)`.

MichaelMcKinsey1 commented:
[fc53a26](https://github.com/LLNL/RAJAPerf/pull/331/commits/fc53a26f8be55fe3e297cad612ecb3b95cf9ce87) For the doc files I changed as little wording as possible for now. It may be necessary to have stronger wording in regards to running 1 variant & 1 tuning at a time, since multiple tunings at a time makes less sense as we removed the `variant.tuning` node level from the caliper annotations. It also may be a good idea to enforce (in the code) single variant/tuning or file splitting when caliper is turned on.

rhornung67 commented on src/common/KernelBase.hpp:
```diff
@@ -29,9 +29,40 @@
+      std::string gstr = getGroupName(kstr); \
+      std::string vstr = "RAJAPerf"; \
+      CaliMeta(); \
+      CALI_MARK_BEGIN(vstr.c_str()); \
```
This doesn't look right to me. Is 'gstr' the "group" name? If so, then getGroupName() should take a GroupID enum value. Is 'vstr' the "variant" name? If so, then getVariantName() should take a VariantID enum value.

Am I misunderstanding this?

MichaelMcKinsey1 replied:
We specficially moved away from `getVariantName()` to a static string, `RAJAPerf`, so different runs will not have different top-level node names. This fits the standard expected by our tools. This information should instead be collected in the metadata, as shown [here](https://github.com/LLNL/RAJAPerf/pull/331/files#diff-f1b7ede2b59bf264efbe43bc92349deb28c8f49dc87efa11b4882ab6805beb24R582).

rhornung67 replied:
OK. I'm still confused about this. I can ask you about it later sometime.

MichaelMcKinsey1 commented:
> [fc53a26](https://github.com/LLNL/RAJAPerf/pull/331/commits/fc53a26f8be55fe3e297cad612ecb3b95cf9ce87) For the doc files I changed as little wording as possible for now. It may be necessary to have stronger wording in regards to running 1 variant & 1 tuning at a time, since multiple tunings at a time makes less sense as we removed the `variant.tuning` node level from the caliper annotations. It also may be a good idea to enforce (in the code) single variant/tuning or file splitting when caliper is turned on.

Addressed in [c0197bf](https://github.com/LLNL/RAJAPerf/pull/331/commits/c0197bf50e16fb2b27641791c1835feaaa1032f7). Generate a single cali file per variant, tuning combo.

MichaelMcKinsey1 commented:
New commits on #352
